common/djangoapps/custom_settings/tests/test_views.py

```diff
@@ -0,0 +1,110 @@
+from datetime import datetime
+from json import dumps
+
+from django.http import HttpResponse
+from django.urls import reverse
+from mock import patch
+from opaque_keys.edx.keys import CourseKey
+from pytz import UTC
+from rest_framework.status import HTTP_200_OK, HTTP_400_BAD_REQUEST, HTTP_405_METHOD_NOT_ALLOWED
+
+from custom_settings.models import CustomSettings
+from custom_settings.tests.factories import CustomSettingsFactory
+from openedx.features.philu_utils.tests.mixins import PhiluThemeMixin
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class CustomSettingsTestCase(PhiluThemeMixin, ModuleStoreTestCase):
+
+    def setUp(self):
+        super(CustomSettingsTestCase, self).setUp()
+        self.client.login(username=self.user.username, password=self.user_password)
+        self.course = CourseFactory.create(
+            org='test',
+            course='mth',
+            run='101',
+            enrollment_start=datetime(2013, 1, 1),
+            enrollment_end=datetime(2014, 1, 1),
+            start=datetime(2013, 1, 1),
+            end=datetime(2030, 1, 1),
+            emit_signals=True,
+        )
+        course_id = 'course-v1:test+mth+101'
+        self.course_key = CourseKey.from_string(course_id)
+        self.url = reverse('custom_settings', kwargs={'course_key_string': course_id})
+        CustomSettingsFactory(id=self.course_key, tags='tag1, tag2', course_short_id=200)
+
+    @patch('custom_settings.views.get_course_and_check_access')
+    def test_course_custom_settings_course_settings_dose_not_exist(self, mock_get_course_and_check_access):
+        """Assert that view throw 400 status when settings are fetched for a course which does not exists"""
+        url = reverse('custom_settings', kwargs={'course_key_string': 'course-v1:test+test+123'})
+        self.assertEqual(self.client.get(url, HTTP_ACCEPT='text/html').status_code, HTTP_400_BAD_REQUEST)
+
+    @patch('custom_settings.views.render_to_response')
+    @patch('custom_settings.views.get_course_and_check_access')
+    @patch('custom_settings.views.get_course_open_date_from_settings')
+    def test_course_custom_settings_get_request(self, mock_get_course_open_date_from_settings,
+                                                mock_get_course_and_check_access,
+                                                mock_render_to_response):
+        """Assert that page show custom settings successfully"""
```

Comment:
*shows

Reply:
Any particular reason for this

Reply:
Grammar. Basically when you write "Page show custom settings", it implies that you are ordering 'Page' to show something. However, if you are stating an event, you'd write it as "Page shows custom settings...". Therefore, the grammatically correct version is:

"Assert that page shows custom settings successfully"

Reply:
For a more realistic example, if I say:

"Zahid, show me your new phone." Here, I'm requesting you to show me your phone.

On the other hand, if I say:

"Zahid shows his new phone." Here I'm stating an event. The two are different.